src/main/scala/nl.knaw.dans.easy.validatebag/rules/metadata/package.scala

```diff
@@ -41,10 +42,14 @@ package object metadata extends DebugEnhancedLogging {
   val allowedFilesXmlNamespaces = List(dcNamespace, dctermsNamespace)
   val allowedAccessRights = List("ANONYMOUS", "RESTRICTED_REQUEST", "NONE")
 
-  val doiPattern: Regex = raw"^10(\.\d+)+/.+".r
+  val doiPattern: Regex = "^10(\\.\\d+)+/.+".r
+  val doiUrlPattern: Regex = "^((https?://(dx\\.)?)?doi\\.org/(urn:)?(doi:)?)?10(\\.\\d+)+/.+".r
+  val urnPattern: Regex = "^urn:[A-Za-z0-9][A-Za-z0-9-]{0,31}:[a-z0-9()+,\\-\\\\.:=@;$_!*'%/?#]+$".r
 
   val daiPrefix = "info:eu-repo/dai/nl/"
 
+  val urlProtocols = List("http", "https")
```

Comment:
Why not match with a regexp 'https?://.*'

Reply:
Just noticed this indeed. Could be a nice improvement. However, since we're checking on protocols/schemes, I'm divided between your solution and `uri.getScheme` https://github.com/DANS-KNAW/easy-validate-dans-bag/blob/d51c8fe4ffd900c7aa96f020bf6287a1812a679d/src/main/scala/nl.knaw.dans.easy.validatebag/rules/metadata/package.scala#L401-L406

suggestions?